app/response/utils.py

```diff
@@ -216,6 +216,42 @@ def add_closing(request_id, reason_ids, email_content):
         _send_response_email(request_id, privacy, email_content)
 
 
+def add_reopening(request_id, date, tz_name, email_content):
+    """
+    Create and store a re-opened-determination for the specified request and update the request accordingly.
+
+    :param request_id: FOIL request ID
+    :param date: string of new date of request completion
+    :param tz_name: client's timezone name
+    :param email_content: email body associated with the reopened request
+
+    """
+    if Requests.query.filter_by(id=request_id).one().status == request_status.CLOSED:
+        new_due_date = process_due_date(datetime.strptime(date, '%Y-%m-%d'), tz_name)
+        update_object(
```

Comment:
I pulled from your branch and github does not appear to be reporting anything incorrect, so I ask again:
Why don't you consider updating the Request once with the new due date, status, and agency description release date all at once?